src/unit.rs

```diff
@@ -233,6 +233,10 @@ impl Unit {
 			self.data.enemy_upgrades.lock_read()
 		}
 	}
+	/// Name of the unit
+	pub fn name(&self) -> String {
+		self.type_data().map_or(String::new(), |data| data.name.clone())
```

Comment:
Cloning name is unnecessary here, It's possible to just return `&str`.

Reply:
Sorry, brain fart on my side. Fixed now